examples/vision/siamesenetwork.py

```diff
@@ -0,0 +1,417 @@
+"""
+Title: Siamese Network with Triplet Loss
+Author: [Hazem Essam](https://twitter.com/hazemessamm) and [Santiago L. Valdarrama](https://twitter.com/svpino)
+Date created: 2021/03/13
+Last modified: 2021/03/22
+Description: Siamese network with tf.data and custom training loop.
+"""
+
+
+"""
+TODO: Still need to work on this.
+
+# Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network) is used to solve
+many problems like detecting question duplicates, face recognition by comparing the
+similarity of the inputs by comparing their feature vectors.
+First we need to have a dataset that contains 3 Images, 2 are similar and 1 is different,
+they are called Anchor image, Positive Image and Negative image respectively, we need to
+tell the network that the anchor image and the positive image are similar, we also need
+to tell it that the anchor image and the negative image are NOT similar, we can do that
+by the Triplet Loss Function.
+Triplet Loss function:
+L(Anchor, Positive, Negative) = max((distance(f(Anchor), f(Positive)) -
+distance(f(Anchor), f(Negative)))**2, 0.0)
+Note that the weights are shared which mean that we are only using one model for
+prediction and training
+You can find the dataset here:
+https://drive.google.com/drive/folders/1qQJHA5m-vLMAkBfWEWgGW9n61gC_orHl
```

Comment:
Suggestion: refactor to mention Totally Looks Like dataset here to avoid repeating it again further below.

Links: https://sites.google.com/view/totally-looks-like-dataset
arXiv: https://arxiv.org/pdf/1803.01485v3.pdf (Rosenfeld et al., 2018)

```suggestion
In this example, we will use the [Totally Looks Like](https://sites.google.com/view/totally-looks-like-dataset)
by [Rosenfeld et al., 2018](https://arxiv.org/pdf/1803.01485v3.pdf).
```

@fchollet the original link was to Google Drive - is it any good? We can provide the direct link further below when loading the dataset.